File: Techschool/Techschool/ClientApp/src/app/models/vacations/social-with-children-vacation-form.model.ts
import { VacationFormModel } from './vacation-form.model';
import { SocialWithChildrenVacationModel } from './social-with-children-vacation.model';

export class SocialWithChildrenVacationFormModel extends VacationFormModel {
  childAge: number;
  days: number;
  socialWithChildrenVacations: SocialWithChildrenVacationModel[];
}

File: Techschool/Techschool.BLL/Models/Vacations/SocialWithChildrenVacationFormModel.cs
using System.Collections.Generic;

namespace Techschool.BLL.Models.Vacations
{
    public class SocialWithChildrenVacationFormModel : VacationFormModel
    {
        public int ChildAge { get; set; }
        public int Days { get; set; }
        public IEnumerable<SocialWithChildrenVacationModel> SocialWithChildrenVacations { get; set; }
    }
}
